### DISCUSSION TITLE
useQuery async with immediate refresh causes refetch after some try

### DISCUSSION THREAD

**Discussion by sabasayer** [posva/pinia-colada#141] [Questions]
I have a situation that i need to call an api multiple times in paralel with different params. 
Created a composable that accepts an argument and uses that as key to query and also passes to fetch method

```typescript
export const useWadoSeriesQuery = (url: string) => {
  const { fetchWadoSeries } = usePatientDocumentService();

  return useQuery({
    key: ["wadoSeries", url],
    query: () => fetchWadoSeries(url),
    enabled: !!url,
  });
};

```

created a method so i can call this query in a loop and my expectation was that it won't be called again unless i invalidate the cache

```typescript
export const readWadoSeries = async (
  documents: PatientDocument[],
): Promise<PatientWado[]> => {
  const wadoSeries: PatientWado[] = [];

  for (const item of documents) {
    const { data, error, refresh } = useWadoSeriesQuery(item.retrieveUrl);
    // This if solves the problem
    // if (!data.value) 
    await refresh();
    if (data.value) {
      wadoSeries.push({
        patientID: item.id,
        wado: data.value,
      });
    } else {
      console.error(
        `Failed to fetch WADO series for document ID ${item.id}: Here is the error: ${error.value}`,
      );
    }
  }

  return wadoSeries;
};
```

This method is called in a modal component which is mounted on visible.
I had to use `await refresh()` because if not initial values will be null. But this causes refetching after **some** try even though data is still in the cache.

I found the solution to run refresh only if data is not exists but i dont think this is the right approach.
Maybe api will return empty data or throw error in first try. I can consider all and add more conditions but it gets messy and i din't want it.

Maybe there is a better way.

Here a reproduction link of [stackblitz](https://stackblitz.com/edit/github-hid7x55w-vnenzbtx?file=src%2Fcomponents%2FTheWelcome.vue) with similar setup.

**Comment by niklaswolf** [comment]
This is also somewhat similar to what I'm trying to achieve: https://github.com/posva/pinia-colada/discussions/100
Basically it is using [external properties for a query](https://pinia-colada.esm.dev/guide/queries.html#Using-External-Properties-in-Queries),  but I'd really like to have a composable like your `useWadoSeriesQuery` here, where I can pass the external value as a parameter, too.
